### ISSUE TITLE
Cannot update slider on Manual page because of Nextion library design

### ISSUE THREAD

**Issue by ccrause** [ccrause/syringepump#18]
The safeRun function needs to call nexLoop in the while loop to listen for the appropriate pop event. If anything is written to the Nextion, such as progressBar1.setValue() the sendCommand function will first empty the serial receive buffer, possibly discarding the pop event.

This is a design flaw in the library, basically it is not designed to handle asynchronous replies. Functions in the library which discards unexpected data: 
- recvRetNumber
- recvRetString
- recvRetCommandFinished
- nexLoop

So basically every receive function in NexHardware.cpp
